connectors/sources/onedrive.py

```diff
@@ -656,7 +656,12 @@ async def get_entity_permission(self, user_id, file_id):
             user_id=user_id, file_id=file_id
         ):
             if identity := permission.get("grantedToV2"):
-                permissions.append(_prefix_user_id(identity.get("user").get("id")))
+                identity_id = identity.get("user", {}).get("id")
+                if identity_id:
+                    permissions.append(_prefix_user_id(identity_id))
+                else:
+                    log = f"No user id found for user {user_id} for file {file_id} in `grantedToV2`"
+                    self._logger.warning(log)
```

Comment:
So user won't have access to this file?

Reply:
Good point, perhaps this log should be re-thought.
Instead maybe log when the fields `user` or `group` fields are present but they don't have an `id`. WDYT?

Reply:
I've changed it to only log a warning if the response has `user` or `group` and no nested `id`